plugins/linux-v4l2/v4l2-output.c

```diff
@@ -0,0 +1,179 @@
+#include <obs-module.h>
+#include <util/platform.h>
+#include <linux/videodev2.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#define MAX_DEVICES 64
+
+struct virtualcam_data {
+	obs_output_t *output;
+	int device;
+	uint32_t frame_size;
+};
+
+static const char *virtualcam_name(void *unused)
+{
+	UNUSED_PARAMETER(unused);
+	return "Virtual Camera Output";
+}
+
+static void virtualcam_destroy(void *data)
+{
+	struct virtualcam_data *vcam = (struct virtualcam_data *)data;
+	close(vcam->device);
+	bfree(data);
+}
+
+static bool loopback_module_loaded()
+{
+	bool loaded = false;
+
+	char temp[512];
+
+	FILE *fp = fopen("/proc/modules", "r");
+
+	if (!fp)
+		return false;
+
+	while (fgets(temp, 512, fp)) {
+		if (strstr(temp, "v4l2loopback")) {
+			loaded = true;
+			break;
+		}
+	}
+
+	if (fp)
+		fclose(fp);
+
+	return loaded;
+}
+
+static int loopback_module_load()
+{
+	return system(
+		"pkexec modprobe v4l2loopback exclusive_caps=1 card_label='OBS Virtual Camera' && sleep 0.5");
+}
```

Comment:
Is there a reason the module options are passed via this command line, rather than relying on a file in `/etc/modprobe.d/`? The second would be easier for a user to configure if they perhaps wanted to try without the exclusive_caps (needed for some apps).

Reply:
If the user wants to configure the module themselves they should load it themselves.

Reply:
FWIW on OpenSUSE this fails with

> Cannot run program modprobe: No such file or directory

Specifying the absolute path works, e.g. `pkexec /usr/sbin/modprobe v4l2loopback`

Reply:
Newest update now uses /usr/sbin/modprobe.

Reply:
```
% ls -la /usr/sbin/modprobe /sbin/modprobe
ls: cannot access '/usr/sbin/modprobe': No such file or directory
lrwxrwxrwx 1 root root 9 Feb 10  2019 /sbin/modprobe -> /bin/kmod
```
(debian/stable)

Reply:
Reverted back to just modprobe for now.

Reply:
I think this will always fail on some distributions, as they vary on where the modprobe binary is located and whether the path to the binary is in $PATH. Could the path to modprobe possibly be passed via some compile-time configuration, with a default of just `modprobe`? That would make it easier for distributions (like OpenSUSE) to distribute a version of OBS that works well with their chosen location for modprobe.

Reply:
There is very little to gain from making this configurable. If distributions want to have weird configuration they are free to patch things to match their configuration. The whole point of PATH is to make these things available regardless of where your distribution puts them.